### PR TITLE
Understanding Server-side request forgery attacks 

### LINKED ISSUE #5046: [Security]Understanding server-side request forgery attacks (SSRF attacks)
### Topic Suggestion
Understanding server-side request forgery

### Writing sample(s): 
https://www.section.io/engineering-education/getting-started-with-visual-basic-net/

#### Proposed title of article
Understanding server-side request forgery

#### Proposed article introduction
Server-side request forgery is among the newest additions to the OWASP Top 10 list released this year (2021). It comes in at Number 10 on the list. According to data by OWASP, the incidence rate is relatively low(OWASP,2021). SSRF attacks are most common to Web applications which fetch data from a server to display it to the client. When the application fetches the resources without validating the supplied URL, then it becomes vulnerable to SSRF. An attacker can alter the application’s request and send a malicious request.

#### Key takeaways 
- Overview of SSRF attacks
- Testing for  SSRF 
- Impacts of SSRF attacks
- How to prevent SSRF attacks

#### Article quality
The article will be beginner friendly and also the definitions on the article will short and easy to understand compared to other blogs.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### PR DESCRIPTION
**Pull request by Dyrstiu**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #5046

### CODE REVIEW

**Comment by hectorkambow** [comment]
@section-engineering-education/peer-reviewers

is anyone available to jump on this PR ASAP.

Thank you

**Comment by hectorkambow** [comment]
@Dyrstiu 
Good afternoon and thank you for submitting your article for publishing.
With the recent increase in article volume out team has been working diligently to review every article. Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Comment by geoffrey45** [comment]
@Dyrstiu I will hop on this one.

cc: @hectorkambow

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by geoffrey45** [comment]
@Dyrstiu Please be sure to check the edits that I have made to your article to improve on your future articles.

File: content/articles/understanding-server-side-request-forgery-attacks/index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+### Testing for SSRF attacks.
+We can test our applications for SSRF vulnerabilities in a couple of ways.
+
+The most recommended one is manual code review to check whether the URL inputs are being validated.
+
+Sometimes, we might find that we don’t have access to the source code of our applications, but this should not hinder us from testing our applications.
+
+To show how we can test for SSRF in cases where you don't have access to the source code, I will use Labs from [Portswigger](https://portswigger.net/web-security/ssrf). We will look at a lab for a local server.
+
```
Comment (geoffrey45) [inline content/articles/understanding-server-side-request-forgery-attacks/index.md:42]: @Dyrstiu Add the link to the lab in use for your demonstrations.

Comment (geoffrey45) [reply content/articles/understanding-server-side-request-forgery-attacks/index.md:42]: @Dyrstiu

Comment (Dyrstiu) [reply content/articles/understanding-server-side-request-forgery-attacks/index.md:42]: The link attached is for the labs by portswigger

Comment (geoffrey45) [reply content/articles/understanding-server-side-request-forgery-attacks/index.md:42]: The link points to a generic SSRF page. I'm talking about the lab itself.

https://portswigger.net/web-security/ssrf/lab-basic-ssrf-against-localhost

Comment (Dyrstiu) [reply content/articles/understanding-server-side-request-forgery-attacks/index.md]: @geoffrey45 I have noticed the error and made the corrections .thank you

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**